check/src/typecheck.rs

```diff
@@ -289,11 +289,13 @@ impl<'a> Typecheck<'a> {
 
     fn stack_type(&mut self, id: Symbol, alias: &Alias<Symbol, ArcType>) {
         // Insert variant constructors into the local scope
-        if let Type::Variant(ref row) = *alias.typ {
-            for field in row.row_iter().cloned() {
-                let symbol = self.symbols.symbol(field.name.as_ref());
-                self.original_symbols.insert(symbol, field.name.clone());
-                self.stack_var(field.name, field.typ);
+        if let Type::Forall(_, ref typ) = *alias.typ {
```

Comment:
My bad, was just testing a theory and never fixed it throughly. It should handle both cases.